### PR TITLE
Make crit people not get hit by bullets (unless you aim at them)

### LINKED ISSUE #21497: Bullets/projectiles shouldn't hit people on the floor in crit, unless they're aimed at
Based on this discussion in discord (see pinned post): https://discord.com/channels/310555209753690112/1171134829253906432

SPECIFICALLY _**on the floor**_ and _**in crit**_

If they're in a chair/vehicle/etc and in crit it should 100% hit still

### PR DESCRIPTION
**Pull request by notquitehadouken**
## About the PR

You no longer hit people in crit (or dead) with bullets unless you aim at them
Closes #21497 

"Aim at them" means hover your mouse over them when you fire.
This works with bullets and hitscan (laser), but not spears.

## Why / Balance

See #21497 

## Media

- [ ] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

**Changelog**

:cl:
- tweak: You no longer shoot crit people unless you aim at them specifically.

### CODE REVIEW

**Comment by notquitehadouken** [comment]
it works but no media because i had to go the instant i made it work

File: Content.Server/Projectiles/ProjectileSystem.cs [hunk 1]
```diff
@@ -33,6 +37,13 @@ private void OnStartCollide(EntityUid uid, ProjectileComponent component, ref St
             return;
 
         var target = args.OtherEntity;
+
+        if (component.CursorTarget != target
+            && TryComp(target, out MobStateComponent? targetState)
+            && (targetState.CurrentState == MobState.Critical || targetState.CurrentState == MobState.Dead)
+            && (!TryComp(target, out BuckleComponent? buckleComponent) || !buckleComponent.Buckled))
+            return;
+
```
Comment (metalgearsloth) [inline Content.Server/Projectiles/ProjectileSystem.cs:46]: Shouldn't be hardcoded to mobstate.

**Review by metalgearsloth — CHANGES_REQUESTED** [review]
Isn't the problem the collision layer is different so they never collide? I didn't seed any code that changes it.

File: Content.Server/Projectiles/ProjectileSystem.cs (hunk 1 above)
Comment (notquitehadouken) [reply Content.Server/Projectiles/ProjectileSystem.cs]: i've been thinking about this for 2 days now and i have no idea what it means

**Comment by notquitehadouken** [comment]
fhuck it im opening it again

File: Content.Server/Projectiles/ProjectileSystem.cs (hunk 1 above)
Comment (notquitehadouken) [reply Content.Server/Projectiles/ProjectileSystem.cs]: @metalgearsloth

**Comment by notquitehadouken** [comment]
ok so i had a fucking epiphany and this should be done by tomorrow

**Comment by notquitehadouken** [comment]
it was not done by tomorrow

**Comment by JoeHammad1844** [comment]
does this work right with shotguns?

**Comment by JoeHammad1844** [comment]
"this should be done by tomorrow" is the new "just a week away"

**Comment by notquitehadouken** [comment]
this shit stale ahh, i'm starting over

**Comment by QuietlyWhisper** [comment]
Hope this can come back, it's important for ranged combat and I'm still seeing bodies soaking up shots needlessly and being used as shields somehow

**Comment by notquitehadouken** [comment]
> Hope this can come back, it's important for ranged combat and I'm still seeing bodies soaking up shots needlessly and being used as shields somehow

soon :trollface:

**Comment by Arendian** [comment]
Is it allright with you if I open a PR to fix this issue? It started as a fix to fire extinguishers not extinguishing dead people, and ended up as something that fixes this too.

**Pull request closed without merging**